### ISSUE TITLE
API: Empty name (`""`) accepted as a valid name while creating environments

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#576]
**Describe the bug**

The API is allowing us to create environments with empty names. 

**To Reproduce**
1. Open bruno
2. Make sure you have created a project
3. Try to create an environment from "Create Environment" request
4. Set the name to an empty string ("")
5. Notice that the environment gets created

**Expected behavior**

No entity should be allowed to have an empty name.

**Screenshots**
![image](https://github.com/user-attachments/assets/ba27e99b-8274-4d05-b95b-c85809eed930)

**Additional context**

This error could be mitigated by enforcing `@IsNotEmpty()` property from `class-validator`.

 Additionally, we would also like to enforce this role to each and every entity name: Workspace, Project, Secret, Variable, Integration, API Key, Workspace Role. Also include tests to verify the same.

**Comment by zaheer-Khan3260** [comment]
Hey @rajdip-b, I am Searching Open source projects for contributing and I found out this projects i love to work on this issue. Can you assign it to me?

**Comment by rajdip-b** [comment]
Sure thing! Do go through our contribution guidelines and our [docs](https://docs.keyshade.xyz) before you get started. Also, don't hesitate to reach out to us in case you run into any issues. Happy hacking!

**Comment by zaheer-Khan3260** [comment]
Hey @rajdip-b, I fix this bug but i am not able to test the changes because the command "pnpm run dev:api" throw error even after set all required environment variable. can you assist me further?

**Comment by Allan2000-Git** [comment]
Can you send the screenshot of your terminal where error occurs?

**Comment by zaheer-Khan3260** [comment]
yeah sure. 
![Screenshot 2024-12-05 224531](https://github.com/user-attachments/assets/2ce57ca2-8d61-4e6c-9b7c-50e6c073b1a8)

**Comment by Allan2000-Git** [comment]
Did you run the following?
pnpm run db:generate-types
pnpm run db:deploy-migrations

**Comment by zaheer-Khan3260** [comment]
yes i follow all the prerequisite that is mantion in the docs and defne all the required env variable.

**Comment by rajdip-b** [comment]
> yeah sure. ![Screenshot 2024-12-05 224531](https://private-user-images.githubusercontent.com/150288870/392922322-2ce57ca2-8d61-4e6c-9b7c-50e6c073b1a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0NzA5MTcsIm5iZiI6MTczMzQ3MDYxNywicGF0aCI6Ii8xNTAyODg4NzAvMzkyOTIyMzIyLTJjZTU3Y2EyLThkNjEtNGU2Yy05YjdjLTUwZTZjMDczYjFhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNlQwNzM2NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYTExNDI5ODM3OTBhYzI4ODJlOGIxOGRjZTVmYjg4Zjg0MzNlZDdhY2ZiYjBiOTVlOWUyZWFjYWQwOWJiMmMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dJ5kkJqlW4pyU2FPDQQMGOUEq-El4llGuw7XFB7B290)

Wow this is new

**Comment by rajdip-b** [comment]
would you be able to share us your env? please remove the sensitive stuff like api keys and passwords

**Comment by zaheer-Khan3260** [comment]
https://github.com/user-attachments/assets/0cef8fa6-2366-45e0-9a61-209042014de1

**Comment by poswalsameer** [comment]
> yeah sure. ![Screenshot 2024-12-05 224531](https://private-user-images.githubusercontent.com/150288870/392922322-2ce57ca2-8d61-4e6c-9b7c-50e6c073b1a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0ODU3MDYsIm5iZiI6MTczMzQ4NTQwNiwicGF0aCI6Ii8xNTAyODg4NzAvMzkyOTIyMzIyLTJjZTU3Y2EyLThkNjEtNGU2Yy05YjdjLTUwZTZjMDczYjFhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNlQxMTQzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTUxODNlNGRhZmM0MWY2NGRjZmUxOWIwYjAxMTMzNjBjZGFmNTQxZjFkNDNkYjMxMmFjYjdiZWNmNTI3Yzc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HeXph4lLpNkG5rLCY97W1Np8Zs2n8bBuyQ5USzfw39c)

Have you spun up the docker container correctly? Are you able to see minio in your docker engine?

**Comment by rajdip-b** [comment]
Minio wont be an issue. Neither would be sentry. That's fine.

**Comment by rajdip-b** [comment]
> Screen.Recording.2024-12-06.163634.mp4

It's the redis url that is at fault.

**Comment by rajdip-b** [comment]
You won't need to use your own cluster , we already provide that via docker compose.

**Comment by rajdip-b** [comment]
![image](https://github.com/user-attachments/assets/50ebb3d8-2907-44b1-a64c-f6310896ec83)
This is what your Redis URL should look like.

**Comment by zaheer-Khan3260** [comment]
Ooh okay

**Comment by zaheer-Khan3260** [comment]
Hey @rajdip-b can you please tell which SMTP provider you are using? If you are comfortable for sharing.
